tests/metal/texture.slang

```diff
@@ -0,0 +1,672 @@
+        && all(Tv(1) == t2DArray.SampleBias(samplerState, float3(u, u, 0), float(-1), int2(1, 1)))
+
+        // ===================================
+        //  T SampleLevel()
+        // ===================================
+
+#if defined(NEED_TO_TEST_FOR_METAL_CAPABILITY)
+        // Metal doesn't support LOD for 1D texture
+        && all(Tv(1) == t1D.SampleLevel(samplerState, u, 0))
+        && all(Tv(1) == t1DArray.SampleLevel(samplerState, float2(u, 0), 0))
+#endif
+
+        // METAL: t2D{{.*}}.sample({{.*}} level(
+        // METALLIB: call {{.*}}.sample_texture_2d.v4f32(
+        && all(Tv(1) == t2D.SampleLevel(samplerState, float2(u, u), 0))
+
+        // METAL: t3D{{.*}}.sample({{.*}} level(
+        // METALLIB: call {{.*}}.sample_texture_3d.v4f32(
+        && all(Tv(1) == t3D.SampleLevel(samplerState, float3(u, u, u), 0))
+
+#if defined(TEST_AFTER_FIXING_CAPABILITY_PROBLEM)
+        // M-ETAL: tCube{{.*}}.sample({{.*}} level(
+        // M-ETALLIB: call {{.*}}.sample_texture_cube.v4f32(
+        && all(Tv(1) == tCube.SampleLevel(samplerState, normalize(float3(u, 1 - u, u)), 0))
+#endif
+
+        // METAL: t2DArray{{.*}}.sample({{.*}} level(
+        // METALLIB: call {{.*}}.sample_texture_2d_array.v4f32(
+        && all(Tv(1) == t2DArray.SampleLevel(samplerState, float3(u, u, 0), 0))
+
+        // METAL: tCubeArray{{.*}}.sample({{.*}} level(
+        // METALLIB: call {{.*}}.sample_texture_cube_array.v4f32(
+        && all(Tv(1) == tCubeArray.SampleLevel(samplerState, float4(u, u, u, 0), 0))
+
+        // Offset variant
+
+#if defined(NEED_TO_TEST_FOR_METAL_CAPABILITY)
+        // Metal doesn't support LOD for 1D texture
+        && all(Tv(1) == t1D.SampleLevel(samplerState, u2, 0, 1))
+        && all(Tv(1) == t1DArray.SampleLevel(samplerState, float2(u2, 0), 0, 1))
+#endif
+
+        // METAL: t2D{{.*}}.sample({{.*}} level(
+        // METALLIB: call {{.*}}.sample_texture_2d.v4f32(
+        && all(Tv(1) == t2D.SampleLevel(samplerState, float2(u, u), 0, int2(1, 1)))
+
+        // METAL: t3D{{.*}}.sample({{.*}} level(
+        // METALLIB: call {{.*}}.sample_texture_3d.v4f32(
+        && all(Tv(1) == t3D.SampleLevel(samplerState, float3(u, u, u), 0, int3(1, 1, 1)))
+
+        // METAL: t2DArray{{.*}}.sample({{.*}} level(
+        // METALLIB: call {{.*}}.sample_texture_2d_array.v4f32(
+        && all(Tv(1) == t2DArray.SampleLevel(samplerState, float3(u, u, 0), 0, int2(1, 1)))
+
+        // ==================
+        // float SampleCmp()
+        // ==================
+
+        // METAL: d2D{{.*}}.sample_compare(
+        // METALLIB: call {{.*}}.sample_compare_depth_2d.f32(
+        && float(1) == d2D.SampleCmp(shadowSampler, float2(u, u), 0)
+
+#if defined(TEST_AFTER_FIXING_CAPABILITY_PROBLEM)
+        // M-ETAL: d2DArray{{.*}}.sample_compare(
+        // M-ETALLIB: call {{.*}}.sample_compare_depth_2d_array.f32(
+        && float(1) == d2DArray.SampleCmp(shadowSampler, normalize(float3(u, 1 - u, u)), 0)
+#endif
```

Comment:
@ArielG-NV ,can you take a look at these two cases and give me some help on why the second one doesn't work with the capability system?
Both of them use the same signature,
```
    [__readNone]
    [ForceInline]
    [require(glsl_hlsl_metal_spirv, texture_shadowlod)]
    float SampleCmp(SamplerComparisonState s, vector<float, Shape.dimensions+isArray> location, float compareValue)
```
How come one of them works and the other one doesn't?
When the issue occurs, the exact error message is following,
```
tests/metal/texture.slang(661): error 36107: entrypoint 'computeMain' requires capability 'hlsl + fragment | hlsl + compute | hlsl + raygen | hlsl + intersection | hlsl + anyhit | hlsl + closesthit | hlsl + miss | hlsl + callable | glsl | spirv_1_0', which is incompatible with the current compilation target 'metal + compute'.
void computeMain()
     ^~~~~~~~~~~
}
```

Reply:
debugging what is wrong locally, I will update when I figure out what went wrong. 
There were a few issues with the old capability system skipping to check certain capabilities that I found while moving over to the new system, I will check if this is one of these scenarios or if something else happened.

Reply:
I currently only see the error:
```
standard error = {
tests/metal/texture.slang(142): note 99999: an internal error threw an exception while working on code near this location
    t1D.GetDimensions(width);
                     ^
(0): error 99999: Slang compilation aborted due to an exception of class Slang::InternalError: assert failure: (0 <= argIndex) && (argIndex < m_argCount)
}
```
is your local code pushed to your branch (so I may test it)?

Reply:
update: everything works after latest PR